src/server/project.ts

```diff
@@ -2212,6 +2218,20 @@ export abstract class Project implements LanguageServiceHost, ModuleResolutionHo
         return this.noDtsResolutionProject;
     }
 
+    /** @internal */
+    updateTargetFile(rootFile: string, targetFileText: string, pastedText: string): {updatedFile: SourceFile | undefined, updatedProgram: Program | undefined, originalProgram: ts.Program | undefined} {

```

Comment:
Why does other functionality needs to be in the services and why cant it be in projectService instead?